compass/config.py

```diff
@@ -1,119 +1,399 @@
+        if self._combined is None:
+            self._combine()
+        return self._combined.get(section, option)
+
+    def getint(self, section, option):
+        """
+        Get an option integer value for a given section.
+
+        Parameters
+        ----------
+        section : str
+            The name of the config section
+
+        option : str
+            The name of the config option
+
+        Returns
+        -------
+        value : int
+            The value of the config option
+        """
+        if self._combined is None:
+            self._combine()
+        return self._combined.getint(section, option)
+
+    def getfloat(self, section, option):
+        """
+        Get an option float value for a given section.
+
+        Parameters
+        ----------
+        section : str
+            The name of the config section
+
+        option : str
+            The name of the config option
+
+        Returns
+        -------
+        value : float
+            The value of the config option
+        """
+        if self._combined is None:
+            self._combine()
+        return self._combined.getfloat(section, option)
+
+    def getboolean(self, section, option):
+        """
+        Get an option boolean value for a given section.
+
+        Parameters
+        ----------
+        section : str
+            The name of the config section
+
+        option : str
+            The name of the config option
+
+        Returns
+        -------
+        value : bool
+            The value of the config option
+        """
+        if self._combined is None:
+            self._combine()
+        return self._combined.getboolean(section, option)
+
+    def getlist(self, section, option, dtype=str):
+        """
+        Get an option value as a list for a given section.
+
+        Parameters
+        ----------
+        section : str
+            The name of the config section
+
+        option : str
+            The name of the config option
+
+        dtype : {Type[str], Type[int], Type[float]}
+            The type of the elements in the list
+
+        Returns
+        -------
+        value : list
+            The value of the config option parsed into a list
+        """
+        values = self.get(section, option)
+        values = [dtype(value) for value in values.replace(',', ' ').split()]
+        return values
+
+    def has_option(self, section, option):
+        """
+        Whether the given section has the given option
+
+        Parameters
+        ----------
+        section : str
+            The name of the config section
+
+        option : str
+            The name of the config option
+
+        Returns
+        -------
+        found : bool
+            Whether the option was found in the section
+        """
+        if self._combined is None:
+            self._combine()
+        return self._combined.has_option(section, option)
+
+    def set(self, section, option, value=None, comment=''):
```

Comment:
@matthewhoffman, I haven't yet used the `comment` argument anywhere in the code.  I'd like to add a few examples both to make sure it works as expected and because I think config options that are created in the code (as opposed to just having their previous values changed) should ideally have a comment in the resulting file.

Reply:
Okay, I've now done this (and fixed a bug related to it).